<?xml version="1.0"?>
<launch>
    <include file="$(find assignment_2_2023)/launch/sim_w1.launch" />
    
    <!-- Set initial values for des_pos_x and des_pos_y -->
    <param name="des_pos_x" value= "0.0" />
    <param name="des_pos_y" value= "1.0" />
    <!-- Set the size of the averaging window -->
    <param name="avg_window_size" value="10" />
    
    <!-- Start the nodes -->
    <node pkg="assignment_2_2023" type="wall_follow_service.py" name="wall_follower" />
    <node pkg="assignment_2_2023" type="go_to_point_service.py" name="go_to_point"  />
    <node pkg="assignment_2_2023" type="node_b.py" name="last_target_service_node" output="screen"/>
    <node pkg="assignment_2_2023" type="node_c.py" name="distance_from_goal" output="screen"/>
    <node pkg="assignment_2_2023" type="node_a.py" name="robot_controller" output="screen" />
    <node pkg="assignment_2_2023" type="bug_as.py" name="bug_action_service" />
</launch>
